code/modules/chemistry/Reagents-PoisonEtc.dm

```diff
@@ -254,20 +254,35 @@ datum
 			penetrates_skin = 1
 			blob_damage = 5
 			value = 7 // 3 2 1 heat
+			var/counter = 1
 
 			on_mob_life(var/mob/M, var/mult = 1) // -cogwerks. previous version
 				if (!M) M = holder.my_atom
+				if (!counter) counter = 1
 				M.take_toxin_damage(1.5 * mult)
 				if (probmult(8))
 					M.emote("drool")
 				if (prob(10))
 					boutput(M, "<span class='alert'>You cannot breathe!</span>")
-					M.losebreath += (1 * mult)
+					M.losebreath += (mult)
 					M.emote("gasp")
-				if (prob(8))
-					boutput(M, "<span class='alert'>You feel horribly weak.</span>")
-					M.setStatus("stunned", max(M.getStatusDuration("stunned"), 30 * mult))
-					M.take_toxin_damage(2 * mult)
+				switch(counter+= (mult))
+					if (20 to 30)
+						boutput(M, "<span class='alert'>You feel a bit weak.</span>")
+						M.setStatus("stunned", max(M.getStatusDuration("stunned"), 1 SECOND * mult))
+						M.take_toxin_damage(mult)
+						counter = 31
+					if (50 to 60)
+						boutput(M, "<span class='alert'>You feel very weak.</span>")
+						M.setStatus("stunned", max(M.getStatusDuration("stunned"), 2 SECONDS * mult))
+						M.take_toxin_damage(1.5 * mult)
+						counter = 61
+					if (75 to INFINITY)
+						boutput(M, "<span class='alert'>You feel horribly weak.</span>")
+						M.setStatus("stunned", max(M.getStatusDuration("stunned"), 3 SECONDS * mult))
+						M.take_toxin_damage(2 * mult)
+						counter = 65
```

Comment:
Nothing between 30-40 and 60-75?

Reply:
When it reaches any number between 30 and 40 it does the stun, then it jumps to 31 so it can start counting again, its gonna take 20 more cycles to the 50 one, which is where the next stun happens, this is a fix to the specific 20 and 40 ones that generaly got skipped, as you pointed out earlyer, i had experienced that issue when testing, and that is the solution i thought of.